pkg/ring/lifecycler.go

```diff
@@ -511,6 +511,17 @@ func (i *Lifecycler) initRing(ctx context.Context) error {
 			return ringDesc, true, nil
 		}
 
+		// If the ingester is in the JOINING state this means it crashed due to
+		// a failed token transfer or some other reason during startup. We want
+		// to set it back to PENDING in order to start the lifecycle from the
+		// beginning.
+		if ingesterDesc.State == JOINING {
+			level.Warn(util.Logger).Log("msg", "instance found in ring as JOINING, setting to PENDING",
+				"ring", i.RingName)
+			ingesterDesc.State = PENDING
```

Comment:
Do we want to set `ingesterDesc.Tokens = nil` as well?

Reply:
If there are tokens, ingester will reuse them. If it needs more, it will generate more when joining. I think this is fine.